snarkWrapper/contracts/Miximus_ERC20.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity ^0.4.19;
+
+import "./MerkelTree.sol";
+import "./Verifier.sol";
+import "./ERC20/ERC20.sol";
+
+contract Miximus is MerkelTree {
+
+    struct MiximusTree {
+        ERC20 ercInstance;
+        bool instantiated;
+        mapping (bytes32 => bool) roots;
+        mapping (bytes32 => bool) nullifiers;
+    }
+    
+    // erc20 to miximus
+    mapping (address => MiximusTree) public miximusTrees;
+
+    event Withdraw (address); 
+    Verifier public zksnark_verify;
+    function Miximus (address _zksnark_verify) {
+        zksnark_verify = Verifier(_zksnark_verify);
+    }
+
+    function deposit (address _erc20Address, bytes32 leaf) payable  {
+        if(!miximusTrees[_erc20Address].instantiated) {
+            miximusTrees[_erc20Address].ercInstance = ERC20(_erc20Address);
+            miximusTrees[_erc20Address].instantiated = true;
+        }
+        require(miximusTrees[_erc20Address].ercInstance.transferFrom(msg.sender, address(this), 1));
+        insert(leaf);
```

Comment:
@barryWhiteHat I don't know how I should modify that in order to have one MerkelTree per `miximusTree`

Reply:
maybe putting the `MTree` structure into the `miximusTree` such as

```
struct MiximusTree {
        ERC20 ercInstance;
        bool instantiated;
        mapping (bytes32 => bool) roots;
        mapping (bytes32 => bool) nullifiers;
        Mtree mtree;
    }
```
?

Reply:
Yes i think that will work.